src/components/atoms/InputRange/InputRange.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { InputHTMLAttributes, useState, ChangeEvent, useRef, useEffect } from 'react';
+import styled, { css } from 'styled-components';
+import { colors } from '../../../constants/colors';
+import { calculateTrackPosition } from './helpers';
+
+const trackHeight = 8;
+const thumbDiameter = 40;
+
+const TrackStyles = css`
+  box-sizing: border-box;
+  border: none;
+  border-radius: ${trackHeight / 2}px;
+  height: ${trackHeight}px;
+  background: ${colors.neutral.light};
+`;
+
+const ThumbStyles = css`
+  box-sizing: border-box;
+  border: none;
+  border-radius: 50%;
+  width: ${thumbDiameter}px;
+  height: ${thumbDiameter}px;
+  background: #007468;
+`;
+
+const ThumbStylesFocus = css`
+  background: ${colors.base.secondary};
+`;
+
+interface IInput extends InputHTMLAttributes<HTMLInputElement> {
+  trackPosition: number;
+}
+
+const SInputRange = styled.input<IInput>`
+  -webkit-appearance: none;
+  width: 100%;
+  height: ${thumbDiameter}px;
+
+  &::-webkit-slider-runnable-track {
+    ${TrackStyles}
+  }
+  &::-moz-range-track {
+    ${TrackStyles}
+  }
+
+  &::-webkit-slider-thumb {
+    -webkit-appearance: none;
+    margin-top: ${(trackHeight - thumbDiameter) * 0.5}px;
+    ${ThumbStyles}
+  }
+  &::-moz-range-thumb {
+    ${ThumbStyles}
+  }
+
+  &::-webkit-slider-runnable-track {
+    background: -webkit-gradient(
+      linear,
+      0% 0%,
+      100% 0%,
+      color-stop(${({ trackPosition }) => trackPosition}, ${colors.base.primary}),
+      color-stop(${({ trackPosition }) => trackPosition}, ${colors.neutral.medium})
+    );
+  }
+  &::-moz-range-progress {
+    border-radius: ${trackHeight / 2}px;
+    height: ${trackHeight}px;
+    background: ${colors.base.primary};
+  }
+
+  &:focus {
+    outline: none;
+
+    &::-webkit-slider-thumb {
+      ${ThumbStylesFocus}
+    }
+    &::-moz-range-thumb {
+      ${ThumbStylesFocus}
+    }
+  }
+
+  ::-moz-focus-outer {
+    border: 0;
+  }
+`;
+
+interface IInputRange extends InputHTMLAttributes<HTMLInputElement> {
+  defaultValue?: number;
+  value?: number;
+  min?: number;
+  max?: number;
+  step?: number;
+}
+
+const InputRange = ({ min = 0, max = 100, onChange, ...otherProps }: IInputRange) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [trackPosition, setTrackPosition] = useState(0.5);
```

Comment:
shouldn't this be initialized with `defaultValue`?

Reply:
Good point, I forgot to ask about the `0.5` 🤔

Reply:
Why? this is the percentage for the track (from 0 to 1). So, if min is defaulted to 0 and max defaulted to 100, it makes sense to default the trackPosition to 0.5.